hrc_speech_prediction/evaluate_tree_model.py

```diff
@@ -0,0 +1,373 @@
+
+    def test_incremental_learning(self, data_type):
+        classes = np.unique((self.data.labels))
+        score = []
+        for i in range(0, self.X_context.shape[0] - 1):
+            # Train set
+            train_X = self.get_Xs([i])
+            train_Y = self.get_labels([i])
+            # Test set
+            test_X = self.get_Xs([i + 1])
+            test_Y = self.get_labels([i + 1])
+            model = self.model(self.context_actions,
+                               features=data_type,
+                               randomize_context=.25).partial_fit(train_X[0],
+                                                                  train_X[1],
+                                                                  train_Y,
+                                                                  classes)
+            prediction = model.predict(*test_X)
+            score.append(metrics.accuracy_score(
+                test_Y, prediction, normalize=True, sample_weight=None))
+
+        print(score)
+        xaxis = np.array([i for i in range(0, self.X_context.shape[0] - 1)])
+        score = np.array(score)
+
+        xnew = np.linspace(xaxis.min(), xaxis.max(), 1000)
+        score_smooth = spline(xaxis, score,xnew)
+
+        plt.plot(xnew, score_smooth)
+        plt.show()
+
+
+    def _evaluate_on(self, train_idx, test_idx, data_type, speech_eps):
+        # self.check_indices(train_idx, test_idx)
+
+        flat_train_idx = [i for p in train_idx for t in p for i in t]
+        flat_test_idx = [i for p in test_idx for t in p for i in t]
+        test_utters = [list(self.data.utterances)[i] for i in flat_test_idx]
+        self.Us += [i for i in test_utters]
+
+        context_train  = [self.get_labels(t) for p in train_idx for t in p]
+        # Train set
+        speech_train_Xs = self.get_Xs(flat_train_idx)
+        speech_train_Y  = self.get_labels(flat_train_idx)
+
+        # Test set a list of np arrays
+        # each array represents the utterances for one participant
+        test_X = [self.get_Xs(t)[1] for p in test_idx for t in p]
+        test_Y = self.get_labels(flat_test_idx)
+        self.Ys += [i for i in test_Y]
+        # Train
+        speech_model = self.model(
+            self.context_actions,
+            features=data_type,
+            randomize_context=.25, ).fit(speech_train_Xs[0], speech_train_Xs[1],
+                                         speech_train_Y, online=self.online)
+        # Evaluate
+        self.comined_model = tree.Tree(root=tree.Node(),
+                                       speech_model=speech_model)
+        # train context model on state visitations
+        for t in context_train:
+            self.comined_model.add_branch(t)
+
+        return self._paricipant_accuracy_score(test_X, test_Y,
+                                               self.comined_model, test_utters)
+
+    def _paricipant_accuracy_score(self, test_X, labels, model, utters):
+        combined_model = model
```

Comment:
That's why you want a real combined model here, not a _tree_ (see my comment on the `Tree` class).